### PR TITLE
feat: add support for Import Attributes and RegExp Modifiers

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[x] Add something to the core
[ ] Other, please explain:

Closes https://github.com/eslint/eslint/issues/19014
Closes https://github.com/eslint/eslint/issues/19073

#### What changes did you make? (Give an overview)

* Updated eslint/js dependencies: `espree`, `eslint-scope`, `eslint-visitor-keys`.
* Updated the following naming convention to ignore import attribute keys in both static and dynamic imports: `camelcase`, `id-denylist`, `id-length`, `id-match`.
* Added tests for `no-underscore-dangle` to verify that it ignores import attribute keys (this rule actually ignores object properties except methods).
* Added tests for `astUtils.needsPrecedingSemicolon` (since there are no unit tests, I added tests for the `no-array-constructor` rule) to verify it returns `false` for nodes after import/export declarations with attributes.
* Added tests for `prefer-regex-literals` and `prefer-named-capture-group` rules to verify behavior with regex pattern modifiers.

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Browser test is now constantly failing in CI:

```
 "concise" Reporter:
========= Your concise report ==========
undefined
❌ Failed to setup tests, no tests found
```

It works for me locally.

@christian-bromann any ideas what might be causing this?

**Comment by christian-bromann** [comment]
> any ideas what might be causing this?

Yes, we unfortunately pushed a buggy version. I triggered a new release but atm it seems like GitHub Actions experiences degraded performance.

**Review by nzakas — COMMENTED** [review]
Overall LGTM. Just one spot for a bit of refactoring.

File: lib/rules/utils/ast-utils.js [hunk 1]
```diff
@@ -1132,6 +1132,47 @@ let needsPrecedingSemicolon;
...
+    }
+
+    // dynamic import
+    if (
+        parent.type === "Property" &&
+        !parent.computed &&
+        (parent.key === node || parent.value === node && parent.shorthand && !parent.method) &&
+        parent.parent.type === "ObjectExpression"
+    ) {
+        const objectExpression = parent.parent;
```
[the remark this answers is not in the document]
Comment (mdjermanovic) [reply lib/rules/utils/ast-utils.js:1155]: `objectExpression` is used here, on the right side:

https://github.com/eslint/eslint/blob/263a7e5245d65d441878c058d079bb385b7555de/lib/rules/utils/ast-utils.js#L1159

[the remark this answers is not in the document]
Comment (mdjermanovic) [reply lib/rules/utils/ast-utils.js:1155]: Updated in https://github.com/eslint/eslint/pull/19076/commits/474a29c0c494ce205337ff4d5c87c22f53c0d080

**Review by nzakas — APPROVED** [review]
LGTM. Giving another day so others have a chance to review.

**Pull request merged**